src/lib/yang/binary.lua

```diff
@@ -244,6 +244,11 @@ local function data_emitter(production)
             stream:write_stringref('stringref')
             stream:write_stringref(data)
          end
+      elseif primitive_type == 'empty' then
+         return function (data, stream)
+            stream:write_stringref('stringref')
+            stream:write_stringref('')
```

Comment:
I don't think this is right.  The idea here is that we have what's effectively a boolean flag; if the incoming "data" arg is non-nil then this leaf is present, and otherwise it's not.  However when this data is deserialized, it will always have the value `''`, which is true, even if `data` was nil!

What if instead we do:
```lua
stream:write_stringref('flag')
string:write_uint32(data and 1 or 0)
```

Then we add a case around like 421:

```lua
function readers.flag()
   if stream:read_uint32() ~= 0 then return true end
   return nil
end
```